### ISSUE TITLE
[CoE Starter Kit - BUG] Solution "Center of Excellence - Core Components" failed to import: ImportAsHolding failed with exception :The account used to import the dataflow 'CoE BYODL Model Driven Apps' does not match the dataflow owner account.

### ISSUE THREAD

**Issue by PetarHr** [microsoft/coe-starter-kit#9242]
### Does this bug already exist in our backlog?

- [x] I have checked and confirm this is a new bug.

### Describe the issue

Hi All, 

While trying to upgrade from version 4.37 to 4.45, we get the following error: **Solution "Center of Excellence - Core Components" failed to import: ImportAsHolding failed with exception :The account used to import the dataflow 'CoE BYODL Model Driven Apps' does not match the dataflow owner account. Please change the dataflow ownership or import the dataflow using the dataflow owner's account.**

We're always using the same account for the import and this is the first time we're seeing this issue. 

I've completely removed the "CoE BYODL Model Driven Apps" dataflow via Default solution, but that didn't help. 

How can this be resolved?

### Expected Behavior

Successful upgrade without errors.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.45

### What app or flow are you having the issue with?

CoE BYODL Model Driven Apps dataflow

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

1. Import CenterofExcellenceCoreComponents_4_45_managed.zip in your CoE environment
2. Observe the error.

**Comment by Jenefer-Monroe** [comment]
I've not seen that error before either. You say you deleted the dataflow? Well let's just try an over write that should fix it. 

### How to do an overwrite upgrade in the classic UX:
 
Switch to classic
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/690cafcb-ade8-4571-81d5-df70f97f6266)

Import
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/433cd2d8-7e2a-4d35-a86e-cf569b7777f2)

Select Core > Next
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/f232f4b5-783d-4d8c-af94-21fb6dfd91bf)

Next
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/39c84d9b-7ccf-4aff-8861-6c70ed7cb237)

Overwrite customizations > Import
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/7b23463c-2361-4135-81a8-363a96ca23d5)

**Comment by Cb-dev-platform** [comment]
Hello,

We too are also having this problem with our install. The challenge is we are hesitant to overwrite the current installation as we have an unmanaged solution with core components that we have customised. This would result in those being deleted?

Thanks,
Charlie.

**Comment by Cb-dev-platform** [comment]
Hello,

We too are also having this problem with our install. Though ours was regarding the Flows Last Run Date BYODL dataflow. The challenge is we are hesitant to overwrite the current installation as we have an unmanaged solution with core components that we have customised. This would result in those being deleted?

Thanks.

**Comment by Cb-dev-platform** [comment]
Thanks for your reply Jennifer, we always use the same service account to import the CoE Kit. It’s never changed for us which I suspect is the same for the poster above. We have the CoE base solutions that are managed and then an unmanaged solution with the components and customisations within for each part of the kit (Core and Governance/Audit).  We’re also version 4.37 currently. I don’t think this is an issue of our setup, it seems like a platform/kit related issue perhaps?

In using the UI all the Dataflows were owned by the account that imports the kit which makes this issue even stranger. I suspect it’s something to do with how Dataflows are treated as a component. Can you suggest any other fixes or possibly release a solution without them so I can see if the import works? We can’t overwrite the system due to the high level of extensibility we have conducted and so effectively we’re stuck at the moment.

**Comment by Sinder88** [comment]
I am getting the same error. I tried to follow your steps below, the import started but when i came back two hours later, i was signed out and looks like the import failed. Now when i try again i dont even get the Classic Experience to import. So strange :(
![Image](https://github.com/user-attachments/assets/75d04df6-fe5e-4cd8-bed7-e5b0c6f6ad93)

**Comment by tkudya** [comment]
I think this is a general issue/bug I am having similiar issues with Dataflows not related to COEStarterkit. change the ownership seems to help

**Comment by Jenefer-Monroe** [comment]
Thank you @tkudya for the comments. 

@Cb-dev-platform can you please confirm ownership of the dataflows?

**Comment by Sinder88** [comment]
Any idea what the steps are to change the dataflow owner?

**Comment by Jenefer-Monroe** [comment]
Yes, I know how via the UX but unfortunately it requires knowing who owns it currently. 
![Image](https://github.com/user-attachments/assets/b77f6b36-619f-4ff4-914a-40f453ff0508)

You can find who owns it currently though by looking in the product tables
![Image](https://github.com/user-attachments/assets/7854b86b-f1d9-483c-9e40-5c202df7b1be)

**Comment by tkudya** [comment]
the ID its telling relates to an Entra User Object ID so you can search who the owner is in Entra

**Comment by ErikFMAG** [comment]
I’m encountering the same issue when trying to import the "Center of Excellence - Core Components" solution as an upgrade in the new and classic view. Despite the dataflow table showing the correct owner, I continue to receive this error during import.

"Solution 'Center of Excellence - Core Components' failed to import: ImportAsHolding failed with exception: The account (da1a4ceb-4c71-46e8-ad4e-d9068929fddf) used to import the dataflow 'CoE BYODL Model Driven Apps' (0095f9a7-72ad-4587-9b80-963d8f3cd28f) does not match the dataflow owner account (9e143c45-22c7-452e-8e38-34bbc2c479ad). Please change the dataflow ownership or import the dataflow using the dataflow owner's account."

**Comment by Cb-dev-platform** [comment]
@Jenefer-Monroe  - As Erik has pointed out. We are also seeing the same ID 9e414…9ad. I think this is a Platform issue.

**Comment by kumashiro17** [comment]
Hello,

We are also having this problem with our install. We always test it in a DEV env before the PROD env. 
Same user for installation as usual. 
No personalisation.
I'm trying to find who is the user mentionned in the error message.

Thanks

Pat

**Comment by PetarHr** [comment]
Hi All, 

Sorry for the late reply on this. 

I'm currently trying the "Overwrite customizations" option - will let you know how it goes.

As for the error message, the GUIDs of the Dataflow owner appears to be the same as the one mentioned above - 9e143c45-22c7-452e-8e38-34bb.......... It's a GUID that doesn't exist in our tenant, so all points that there is a general package/platform issue upgrading to this version.

**Comment by Sinder88** [comment]
Thanks, my upgrade worked after updating the dataflow ones to the service account doing the upgrade.

**Comment by kumashiro17** [comment]
i don't know if it's a problem, but the rows are duplicated in my **dev** and **prod** environments 

![Image](https://github.com/user-attachments/assets/9f808a84-1427-4de8-aaaa-5c2e12aa30e6)

**Comment by PetarHr** [comment]
So, the "Overwrite customizations" option did not work - it failed with the same error message. However I see that the CoE BYOD Model Driven Apps that I deleted is back in the Core solution.

When I search for the Dataflows to try to re-assign them thru the "Dataflow" tab I don't get any results. (I presume because our service account owner display name starts with a special symbol).
![Image](https://github.com/user-attachments/assets/e463fe83-3512-48d6-adde-c0bf103fd657)

 

Same for us in the Dataflows table - I see duplicate records. 
![Image](https://github.com/user-attachments/assets/a1b8349f-2781-43b5-9ce5-c4e1b662f2b6)

**Comment by kumashiro17** [comment]
> i don't know if it's a problem, but the rows are duplicated in my **dev** and **prod** environments
> 
> ![Image](https://github.com/user-attachments/assets/9f808a84-1427-4de8-aaaa-5c2e12aa30e6)

@Jenefer-Monroe. I realised that the duplicates' date correspond to an import over write made in october to apply a fix you suggested. Will the duplicates be removed when I'll use the **CoE Setup and Upgrade Wizard** ?

**Comment by Jenefer-Monroe** [comment]
I do not think that was the cause of the duplicates. I see these duplicates in environments as well where I have not done that type of upgrade. 
Usually when I see this there are two different owners, one for the person that installed and one what looks like a product user. 
Seems like that is not what you all are seeing. 

Has anyone posted a support ticket yet? If so, you can point them to @Grant-Archibald-MS who is a Microsoft FTE and can help ensure we make quicker traction without confusion about ownership.

**Comment by Cb-dev-platform** [comment]
> I do not think that was the cause of the duplicates. I see these duplicates in environments as well where I have not done that type of upgrade. Usually when I see this there are two different owners, one for the person that installed and one what looks like a product user. Seems like that is not what you all are seeing.
> 
> Has anyone posted a support ticket yet? If so, you can point them to [@Grant-Archibald-MS](https://github.com/Grant-Archibald-MS) who is a Microsoft FTE and can help ensure we make quicker traction without confusion about ownership.

I have raised a ticket for this just now. I think that Power Platform treats dataflows differently as a component.

**Comment by Cb-dev-platform** [comment]
Just to come back on this, the support engineer on my ticket that I raised has confirmed Microsoft are working on a priority ticket internally.

**Comment by Jenefer-Monroe** [comment]
Thank you SOOO much for driving this with support!!!

**Comment by VitorFaim** [comment]
Hello all,

For the ones that are facing this issue, our Product Group shared the following workaround until we get this fixed:

"In the meantime, we found a workaround that helps. Since this issue is caused by unstable metadata, costumers can change the ownership on the dataflow to someone else and then change it back to themselves and then import. that should fix  the issue."

**Comment by Cb-dev-platform** [comment]
Hi all,

This has been fixed for our development environment today. Microsoft support had to push a fix to our environments. I’m not sure if this has been pushed for all customers but I’d suggest you raise a ticket if you’re still experiencing this issue.

@Jenefer-Monroe FYI.

**Comment by kumashiro17** [comment]
Hello @Cb-dev-platform ,

i opened a support thicket at Microsoft. I joined the reply below. It seems that the department where my ticcket went doesn't know about that problem and suggests to open a ticket in the CoE GitHub.. Can you send me an email to kumashiro17@gmail.com with your ticket number and some details I can provide to person assigned to my ticket. Maybe they will be able to redirect my request to the right group. 

Thanks in advance

Pat

> Hi Pat,
>  
> My name is xxxxx and I am the support resource from the CE Platforms team that has been assigned to your incident.
>  
> I understand that there is a CoE fix for an install issue that is being pushed out and you would like to have that for Dev before it goes into Production.
>  
> Unfortunately, CE Platform Support does not work with CoE fixes and your inquiry should be addressed by CoE Support. We only support the core features of Power Platform Admin Center.
>  
> I would recommend either adding on to that existing string with the CoE bug or log your inquiry at aka.ms/coe-starter-kit-issues. CoE support should be able to provide details on how to obtain their fix for Dev prior to Production.
>  
> Let me know if you have any further questions.

**Comment by VitorFaim** [comment]
@kumashiro17 I'm the support engineer that helped @Cb-dev-platform on this incident!

We have a fix as commented above and this will be automatically installed on the next days, if this was not installed yet. Please confirm if you receive yesterday or even today, an update from the solution "Dataflow Solution" otherwise ask the engineer to ping me.

**Comment by Jenefer-Monroe** [comment]
Thank you so much @VitorFaim !!! Please send @Grant-Archibald-MS a note (he's the internal MSFT owner for CoE Starter Kit)
We'd love to be sure you get recognition for being so responsive here!

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team

**Comment by hilda-hk** [comment]
Im facing this issue too. May I know how can I message @VitorFaim and @Grant-Archibald-MS for sharing the ticket number? Thanks.
